libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/StreamingExtensions/StreamingExtensionRequestHandler.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Protocol;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Integration.AspNet.Core.StreamingExtensions
+{
+    internal class StreamingExtensionRequestHandler : RequestHandler
+    {
+        public StreamingExtensionRequestHandler(BotFrameworkStreamingExtensionsAdapter adapter, IBot bot)
+        {
+            this.StreamingAdapter = adapter;
+            this.Bot = bot;
+        }
+
+        private BotFrameworkStreamingExtensionsAdapter StreamingAdapter { get; set; }
+
+        private IBot Bot { get; set; }
+
+        /// <summary>
+        /// Processes incoming requests and returns the response, if any.
+        /// </summary>
+        /// <param name="request">A ReceiveRequest from the connected channel.</param>
+        /// <returns>A response created by the BotFrameworkStreamingExtensionsAdapter.</returns>
+        public override async Task<Response> ProcessRequestAsync(ReceiveRequest request)
+        {
+            var response = new Response();
+
+            var body = request.ReadBodyAsString().Result;
+
+            if (string.IsNullOrEmpty(body))
+            {
+                response.StatusCode = 400;
+                return response;
+            }
+
+            try
+            {
+                var activity = JsonConvert.DeserializeObject<Activity>(body, SerializationSettings.DefaultDeserializationSettings);
```

Comment:
You might want to check that the request is valid (proper verb, path, and has a body...):

if (string.IsNullOrEmpty(request.Verb) || request.Verb.ToUpperInvariant() != "POST")
            {
                response.StatusCode = 405;
                return response;
            }

            var body = await request.ReadBodyAsString().ConfigureAwait(false);

            if (string.IsNullOrEmpty(body) || request.Streams == null || request.Streams.Count == 0)
            {
                // no body
                response.StatusCode = 400;
                return response;
            }

            if (request.Streams[0].Type != "application/json; charset=utf-8")
            {
                response.StatusCode = 406;
                return response;
            }

            var activity = JsonConvert.DeserializeObject<Activity>(body, SerializationSettings.DefaultDeserializationSettings);

Reply:
How's this?
```
            if (string.IsNullOrEmpty(body) || request.Streams?.Count > 0)
            {
                response.StatusCode = 400;
                return response;
            }

            if (request.Streams.Where(x => x.Type != "application/json; charset=utf8").Any())
            {
                response.StatusCode = 406;
                return response;
            }
```